conans/test/integration/toolchains/cmake/cmakedeps/test_cmakedeps.py

```diff
@@ -187,3 +187,70 @@ def package_info(self):
     assert r"spaces=me you" in deps
     assert r"foobar=bazbuz" in deps
     assert r"answer=42" in deps
+
+
+def test_cmakedeps_set_dependency_props_from_consumer():
+    client = TestClient(path_with_spaces=False)
+    bar = textwrap.dedent(r'''
+        from conan import ConanFile
+        class FooConan(ConanFile):
+            settings = "os", "compiler", "build_type", "arch"
+            def package_info(self):
+                self.cpp_info.set_property("cmake_find_mode", "both")
+                self.cpp_info.components["component1"].requires = []
+        ''')
+
+    foo = textwrap.dedent(r'''
+        from conan import ConanFile
+        from conan.tools.cmake import CMakeDeps, cmake_layout
+        class FooConan(ConanFile):
+            settings = "os", "compiler", "build_type", "arch"
+            requires = "bar/1.0"
+            def layout(self):
+                cmake_layout(self)
+            def generate(self):
+                deps = CMakeDeps(self)
+                deps.set_property("bar", "cmake_find_mode", "{find_mode}")
```

Comment:
```suggestion
                deps.set_property(self.dependencies["bar"], "cmake_find_mode", "{find_mode}")
```
This might dissambiguate, I know it is not as clean and nice, just a rough idea, but to consider how to resolve name collisions. Other alternatives:
- ``deps.set_property(...., build_context=True)``
- Allow both ``str`` and ``ConanFileInterface`` from ``self.dependencies`` as argument.
- ?